# config/bluerov_comms.yaml
# Continuously poll PacketBlueRov at 1 Hz and publish as /cmd_vel.
# Also provide /start and /stop Trigger services (no plugins).

pez_comms:
  ros__parameters:

    #################################################################
    # 1) MODEM I/O SETTINGS
    #################################################################
    modem_io:
      port: "/dev/ttyUSB0"
      baud: 9600
      timeout: 0.1

    #################################################################
    # 2) POLL SCHEDULE (1 Hz)
    #################################################################
    schedules:
      - name: bluerov_poll
        loop: true
        steps:
          - name: poll
            duration: 1.0
            encode:
              type: PacketBlueRov
              seq:         0
              vx:          0.0
              vy:          0.0
              vz:          0.0
              wz:          0.0
              svc_pending: 0
              svc_id:      0
              svc_val:     0

    #################################################################
    # 3) INBOUND PACKET HANDLER
    #################################################################
    serial_handlers:
      - packet: PacketBlueRov
        publish:
          topic: "/cmd_vel"
          type:  "geometry_msgs/msg/Twist"
          mapping:
            linear.x:   "vx"
            linear.y:   "vy"
            linear.z:   "vz"
            angular.z:  "wz"

    #################################################################
    # 4) START/STOP SERVICES
    #################################################################
    services:
      - name: "/start"
        srv_type: "std_srvs/srv/Trigger"
        on_request:
          encode:
            type: PacketBlueRov
            seq:         0
            vx:          0.0
            vy:          0.0
            vz:          0.0
            wz:          0.0
            svc_pending: 1
            svc_id:      1
            svc_val:     1
        on_response:
          publish:
            topic: "/start_done"
            type:  "std_msgs/msg/Bool"
            value: resp.success

      - name: "/stop"
        srv_type: "std_srvs/srv/Trigger"
        on_request:
          encode:
            type: PacketBlueRov
            seq:         0
            vx:          0.0
            vy:          0.0
            vz:          0.0
            wz:          0.0
            svc_pending: 1
            svc_id:      2
            svc_val:     0
        on_response:
          publish:
            topic: "/stop_done"
            type:  "std_msgs/msg/Bool"
            value: resp.success
